backend/src/main/kotlin/app/ehrenamtskarte/backend/userdata/webservice/UserImportHandler.kt

```diff
@@ -37,15 +49,33 @@ class UserImportHandler {
 
             BufferedReader(InputStreamReader(file.content())).use { reader ->
                 getCSVParser(reader).use { csvParser ->
-                    importData(csvParser)
+                    importData(csvParser, project.id.value)
                 }
             }
-            ctx.status(200).result("Import successfully completed")
+            ctx.status(200).json(mapOf("message" to "Import successfully completed"))
         } catch (exception: UserImportException) {
-            ctx.status(400).result(exception.message ?: "Import failed")
+            ctx.status(400).json(mapOf("message" to exception.message))
+        } catch (exception: UnauthorizedException) {
+            ctx.status(401).json(mapOf("message" to exception.message))
+        } catch (exception: ForbiddenException) {
+            ctx.status(403).json(mapOf("message" to exception.message))
+        } catch (exception: ProjectNotFoundException) {
+            ctx.status(404).json(mapOf("message" to exception.message))
         } catch (exception: Exception) {
             logger.error("Failed to perform user import", exception)
-            ctx.status(500).result("Internal error occurred")
+            ctx.status(500).json(mapOf("message" to "Internal error occurred"))
+        }
+    }
+
+    private fun authenticate(ctx: Context): ApiTokenEntity {
+        val authHeader = ctx.header("Authorization")?.takeIf { it.startsWith("Bearer ") }
+            ?: throw UnauthorizedException()
+        val token = authHeader.substring(7)
+
+        return transaction {
+            ApiTokenEntity.find { ApiTokens.expirationDate greater LocalDate.now() }
+                .firstOrNull { BCrypt.verifyer().verify(token.toCharArray(), it.token).verified }
+                ?: throw ForbiddenException()
```

Comment:
It turned out that the current hashing does not allow selecting a token by hash from the database, because the hashing function returns a different result for the same token value every time.
To validate a token we need to get 2 values:
- token from auth header
- hash value from the database
So to find the right token in the database, I select all the values and perform the check on each one.